vike/shared/assertDataHookReturn.ts

```diff
@@ -0,0 +1,20 @@
+export { assertDataHookReturn }
+
+import { assertUsage, isPlainObject } from './utils.js'
+import { assertHookReturnedObject } from './assertHookReturnedObject.js'
+import pc from '@brillout/picocolors'
+
+function assertDataHookReturn<Keys extends readonly string[]>(
+  hookReturnValue: unknown,
+  hookFilePath: string
+): asserts hookReturnValue is undefined | null | { [key in Keys[number]]?: unknown } {
+  if (hookReturnValue === undefined || hookReturnValue === null) {
+    return
+  }
+  const errPrefix = `The data() hook defined by ${hookFilePath}` as const
+  assertUsage(
+    isPlainObject(hookReturnValue),
+    `${errPrefix} should return a plain JavaScript object, ${pc.cyan('undefined')}, or ${pc.cyan('null')}`
+  )
+  assertHookReturnedObject(hookReturnValue, [] as const, errPrefix)
```

Comment:
Hi @AurelienLourot, what about these `[] as const`, what should we put inside these array?, Is it ['data'] ?

Reply:
This assertion is wrong actually, good catch. Any key should be allowed here. The idea is that a `data()` hook that looks like

```js
function data(pageContext) {
  // ...
  return {
    foo: 42,
    bar: 'baz'
  }
}
```

will result in a `pageContext` that looks like

```js
{
  data: {
    foo: 42,
    bar: 'baz'
  }
}
```

I'll fix that :pray:

Reply:
(@phonzammi Btw. we added a couple of tickets to https://github.com/vikejs/vike/issues?q=is%3Aissue+is%3Aopen+label%3A%22contribution-welcome+%3Atwo_hearts%3A%22 in case that's something you'd be interested in.)

Reply:
Sure, 👍

Reply:
> Btw. we added a couple of tickets to https://github.com/vikejs/vike/issues?q=is%3Aissue+is%3Aopen+label%3A%22contribution-welcome+%3Atwo_hearts%3A%22 in case that's something you'd be interested in.)

I'll see if I can, I am still junior

Reply:
> I'll see if I can, I am still junior

I you're willing to dig on your own (e.g. trying to become knowledgeable around about the topics of your PR) then it's totally fine. You can have a try and we see how it goes if you want. Looking forward to some PR "sparring" 🦾

Reply:
Thanks @brillout 
i really appreciate that